src/middlewared/middlewared/plugins/reporting.py

```diff
@@ -890,7 +890,7 @@ def setup(self):
             os.makedirs(os.path.join(pwd, 'localhost'))
 
         # Create "${hostname}" -> "localhost" symlink if necessary
-        if hostname != 'localhost':
+        if hostname != 'localhost' and not os.path.exists(os.path.join(pwd, hostname)):
```

Comment:
This should always not exist, there is code above that removes both symlinks and directories.

Reply:
Seconds thought: if someone creates file named `localhost` or `f"{hostname}"` there, our code might crash, we won't create corresponding directories, maybe that was the case?

Reply:
Okay. I will back out this particular change until I can reliably reproduce / have a better idea on what caused the issue.